awx/main/tasks.py

```diff
@@ -790,12 +790,15 @@ def run(self, pk, isolated_host=None, **kwargs):
             safe_env = self.build_safe_env(env, **kwargs)
 
             # handle custom injectors specified on the CredentialType
-            if hasattr(instance, 'all_credentials'):
-                credentials = instance.all_credentials
-            elif hasattr(instance, 'credential'):
-                credentials = [instance.credential]
+            credentials = []
+            if isinstance(instance, Job):
+                credentials = instance.credentials.all()
             else:
-                credentials = []
+                # once other UnifiedJobs (project updates, inventory updates)
+                # move from a .credential -> .credentials model, we can
+                # lose this block
+                credentials = [instance.credential]
```

Comment:
What if this is a WorkflowJob or SystemJob and doesn't have a `credential` attribute?

Reply:
<grumble> <grumble> something something duck typing

Reply:
(fixing this)

Reply:
_fix approved_